x-pack/plugins/apm/public/hooks/use_table_pagination.ts

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  CriteriaWithPagination,
+  EuiTableSortingType,
+  Pagination,
+} from '@elastic/eui';
+import { orderBy } from 'lodash';
+import { useMemo, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import uuid from 'uuid';
+import { fromQuery, toQuery } from '../components/shared/links/url_helpers';
+import { useApmParams } from './use_apm_params';
+
+interface Initials<T extends any[]> {
+  // totalItemCount is not necessary here
+  initialPagination: Omit<Pagination, 'totalItemCount'>;
+  initialSort?: EuiTableSortingType<T[0]>;
+}
+
+type Props<T extends any[]> = {
+  items: T;
+  urlState?: boolean;
+} & Initials<T>;
+
+const PAGE_SIZE_OPTIONS = [10, 25, 50];
+
+export function useTablePagination<T extends any[]>({
```

Comment:
As the hook handles sorting as well it might be good to add this to the name.

Reply:
I called it `useTableSortAndPagination`